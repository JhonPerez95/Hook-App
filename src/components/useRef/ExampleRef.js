import React, { useState } from 'react';
import MultipleCustomHooks from '../useEffect/MultipleCustomHooks';

const ExampleRef = () => {
  const [show, setShow] = useState(false);

  return (
    <>
      <h1> Real Example UseRef</h1>
      <hr />

      {show && <MultipleCustomHooks />}
      <button
        className="btn btn-primary mt-5"
        onClick={() => {
          setShow(!show);
        }}
      >
        Show / hide
      </button>
    </>
  );
};

export default ExampleRef;
