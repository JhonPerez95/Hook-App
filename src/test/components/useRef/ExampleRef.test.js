import React from 'react';
import { shallow } from 'enzyme';
import ExampleRef from '../../../components/useRef/ExampleRef';

describe('Test al component <ExampleRef/>', () => {
  const wrapper = shallow(<ExampleRef />);

  test('Debe mostrarse correctamente el componnete <ExampleRef/>', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
  });

  test('Debe mostrar el component  <MultipleCustomHooks/>', () => {});
  wrapper.find('button').simulate('click');
  expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
});
