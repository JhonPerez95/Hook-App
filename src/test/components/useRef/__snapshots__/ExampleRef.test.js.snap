// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test al component <ExampleRef/> Debe mostrarse correctamente el componnete <ExampleRef/> 1`] = `
<Fragment>
  <h1>
     Real Example UseRef
  </h1>
  <hr />
  <MultipleCustomHooks />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show / hide
  </button>
</Fragment>
`;
